### PR TITLE
Document registries and packaging process

### LINKED ISSUE #734: Document packaging infrastructure
### Problem

`scarb package` is undocumented

### Proposed Solution

Write a reference package on how to use `scarb package` to get package tarballs. Describe all quirks, the process of source files selection, ignore files handling etc.

### Notes

_No response_

### PR DESCRIPTION
**Pull request by szymmis**
Resolves #734 and #901.

### CODE REVIEW

File: website/docs/registries/package-tarball.md
Comment (mkaput) [inline website/docs/registries/package-tarball.md:57]:
```suggestion
File containing info about version control system used in your package.
```

File: website/docs/registries/package-tarball.md
Comment (mkaput) [inline website/docs/registries/package-tarball.md:58]:
```suggestion
Currently only Git is supported.
```

File: website/docs/registries/overview.md
Comment (mkaput) [inline website/docs/registries/overview.md:1]: Please put each sentence in a separate line. This makes diffs in markdown files significantly better because people mostly operate on sentence level when writing prose, while diff works line-by-line.

File: website/docs/registries/overview.md
Comment (mkaput) [inline website/docs/registries/overview.md:3]: Documentation should focus not on what is changing, but on what is state of the art **now**. Saying that Git is currently the default is unnecessary. At the very moment of publishing these docs, registries will be the default, while Git be the past.

It's a different thing that there is no official registry, but this is a temporary caveat, which should be said as a banner or something in my opinion.

File: website/docs/registries/overview.md
Comment (mkaput) [inline website/docs/registries/overview.md:7]: Use [internal links](https://vitepress.dev/guide/markdown#internal-links). Editors will try to maintain these when moving files etc.

File: website/docs/registries/packaging.md
Comment (mkaput) [inline website/docs/registries/packaging.md:8]:
```suggestion
If you are in a Git repository, Scarb will first check if the repo state is clean and error out in case of any changes present in a git working directory. To ignore this check you can use the `--allow-dirty` flag.
```

File: website/docs/registries/packaging.md
Comment (mkaput) [inline website/docs/registries/packaging.md:10]: Consider emphasizing that this is a dangerous operation because you never know

**Review by mkaput — DISMISSED** [review]

**Comment by mkaput** [comment]
We will probably want to merge this after the registry MVP is released

File: website/docs/guides/dependencies.md
Comment (DelevoXDG) [inline website/docs/guides/dependencies.md:10]: I think we can address this in a separate issue, as rewriting / creating a separate guide will take some additional time, and I'm not sure this is necessary to close this PR and make docs available sooner.

However, if you think it's worth it to have this available right away, I can do it as a part of the same PR / stack.

File: website/docs/registries/overview.md
Comment (DelevoXDG) [reply website/docs/registries/overview.md:1]: Addressed in [78d8823](https://github.com/software-mansion/scarb/pull/962/commits/78d8823eac035a7129490616dd92b2755af93708)

File: website/docs/registries/overview.md
Comment (DelevoXDG) [reply website/docs/registries/overview.md]: Addressed now that the registry is available 😄

File: website/docs/registries/packaging.md
Comment (DelevoXDG) [reply website/docs/registries/packaging.md]: Addressed in [e5b3c7b](https://github.com/software-mansion/scarb/pull/962/commits/e5b3c7beea228e2be519bd0a13e907260a84fa30)

File: website/docs/registries/overview.md
Comment (DelevoXDG) [reply website/docs/registries/overview.md]: Sure, wasn't available yet when I was making the change, addressed in [90966b0](https://github.com/software-mansion/scarb/pull/962/commits/90966b089b3014b969a893a55bb62e2d09ab0848)

File: website/docs/registries/custom-registry.md
Comment (DelevoXDG) [reply website/docs/registries/custom-registry.md:28]: Yeah, but I'm not sure we can say it's available in the context of API - it's available from the registry GUI, but there's no API endpoint at the moment and no Scarb command that interacts with that endpoint. As of now, afaiu, this statement
> Currently, isn't used at all, 

still holds

Comment (THenry14) [reply website/docs/registries/custom-registry.md:28]: ah, in that sense, ok thanks :+1:

**Review by THenry14 — APPROVED** [review]
lgtm but please wait for at least one additional approval

File: website/docs/reference/specifying-dependencies.md
Comment (Arcticae) [inline website/docs/reference/specifying-dependencies.md:116]: which is correct, the above or this?:
```suggestion
alexandria_math = "0.1.0"
```

File: website/docs/registries/package-tarball.md
Comment (Arcticae) [inline website/docs/registries/package-tarball.md:58]: It's unclear why it's relevant to include in the tarball at all

File: website/docs/registries/overview.md
Comment (Arcticae) [inline website/docs/registries/overview.md:14]: This should be last here, it seems really exotic to me

File: website/docs/registries/packaging.md
Comment (Arcticae) [inline website/docs/registries/packaging.md:1]: I think `Publishing` is more appropriate, since it's the purpose

File: website/docs/registries/packaging.md
Comment (Arcticae) [inline website/docs/registries/packaging.md:3]: This is a good in-depth description although i think a basic outline first "what" and after this "why"/"how" would make it more clear

File: website/docs/registries/packaging.md
Comment (Arcticae) [inline website/docs/registries/packaging.md:33]: That's a bummer that i learn this only after reading the whole page

File: website/docs/registries/overview.md
Comment (Arcticae) [inline website/docs/registries/overview.md:12]: This should be stated as an info "bubble" above so it's read first, and can be comfortably removed later

File: website/docs/reference/specifying-dependencies.md
Comment (DelevoXDG) [reply website/docs/reference/specifying-dependencies.md:116]: They're equivalent, this explains that mentioned snippet:
```toml
alexandria_math = { version = "0.1.0" }
```
will default to using the official registry if no other keys are specified.

File: website/docs/registries/overview.md
Comment (DelevoXDG) [reply website/docs/registries/overview.md:12]: Can you please clarify what exactly do you mean by "above"?

Would seem strange to introduce one missing feature prior to introducing the registry itself.

Comment (Arcticae) [reply website/docs/registries/overview.md:12]: By above i mean at the top of this doc, under the title

It's a limitation that's really important to understand prior to reading any content about the registry. It tells about the stage that it's at.

It's the most important information here for the user, far more important than telling him it's in development, or that they can explore packages.

File: website/docs/reference/specifying-dependencies.md
Comment (Arcticae) [reply website/docs/reference/specifying-dependencies.md:116]: It's unclear from the doc that the format is interchangeable

File: website/docs/registries/overview.md
Comment (DelevoXDG) [reply website/docs/registries/overview.md]: Yeah, I suppose that makes sense, addressed in a372664c0a24dffc572a231b10077b8de85eed0f

File: website/docs/registries/packaging.md
Comment (DelevoXDG) [reply website/docs/registries/packaging.md]: Addressed in 247cb0390d313743b6832a3ab02a0598a203d224

File: website/docs/registries/overview.md
Comment (DelevoXDG) [reply website/docs/registries/overview.md:12]: > By above i mean at the top of this doc, under the title

While this is not my PR originally, it is my understanding that the idea of these docs was to describe registries as a concept, not exclusively the official registry, which is why (probably) the section is titled *registries* and not *scarbs.xyz*, so I don't think I can really include it at the top of the doc, as `Overview` isn't specifically about the official registry.

> It's the most important information here for the user, far more important than telling him it's in development, or that they can explore packages.

While we can consider adding it as a "bubble" under `Official registry`, I believe the average user is more interested in downloading packages rather than sharing them, and in knowing what they *can do*, not what they can't. 

Imo anyone interesting in publishing their package will probably be willing one extra sentence.

File: website/docs/reference/specifying-dependencies.md
Comment (DelevoXDG) [reply website/docs/reference/specifying-dependencies.md]: Addressed in 62da5d680363497f033ef8816c3717dc04e4679c, please TAL

File: website/docs/registries/package-tarball.md
Comment (DelevoXDG) [reply website/docs/registries/package-tarball.md:58]: Can you please clarify whether it's `VCS.json` itself that concerns you or its presence in the docs?

File: website/docs/registries/packaging.md
Comment (DelevoXDG) [reply website/docs/registries/packaging.md]: Discussed offline, addressed in 854ec0a6f796883eca8f18a1b5db85a854367f43

File: website/docs/registries/packaging.md
Comment (DelevoXDG) [reply website/docs/registries/packaging.md]: IIUC this is only related to the current `scarb publish`, not the entire page, but I've tried to adjust this section a bit to include information about limitation in the warning bubble at the top of the section, along with the information about uploading to the official registry: 9b94262f6fac4c3dc587597bd06027de93414c7b

Please TAL

File: website/docs/registries/package-tarball.md
Comment (Arcticae) [reply website/docs/registries/package-tarball.md:58]: It concerns me that i don't know why the tool includes it in the tarball, what is its' purpose after packaging?

**Review by Arcticae — APPROVED** [review]

File: website/docs/registries/package-tarball.md
Comment (DelevoXDG) [reply website/docs/registries/package-tarball.md:58]: I can't tell you for sure as I wasn't adding these [changes](https://github.com/software-mansion/scarb/pull/827), only documenting the current state of the tool. But, if I were to assume, this mirrors cargo, and will probably come in handy later on when Github-based package is verification introduced.

Not sure we need user to know why this exists, only that this is a reserved file name.

**Comment by DelevoXDG** [comment]
@mkaput 

Can you please approve / TAL, as the requested changes will block the merge? 🙏🏻

File: website/docs/guides/dependencies.md
Comment (DelevoXDG) [reply website/docs/guides/dependencies.md:10]: #1512

File: website/docs/reference/specifying-dependencies.md
Comment (maciektr) [inline website/docs/reference/specifying-dependencies.md:5]: As a reference, this should not be limited to the official registry, but also include a example of using another registry by url.

File: website/docs/registries/publishing.md
Comment (maciektr) [inline website/docs/registries/publishing.md:6]:
```suggestion
## Packaging your package
```
Project may consist of multiple packages

File: website/docs/guides/dependencies.md
Comment (maciektr) [reply website/docs/guides/dependencies.md:10]: IMHO this does not reflect the fact, that registry is still in beta.

Comment (THenry14) [reply website/docs/guides/dependencies.md]: Included beta info in newest commit

File: website/docs/reference/specifying-dependencies.md
Comment (THenry14) [reply website/docs/reference/specifying-dependencies.md]: Removed `official` from the header, and added an example of using custom registry i newest commit

**Pull request merged**
